### ISSUE TITLE
Status of run_results.json nodes are not accurate.

### ISSUE THREAD

**Issue by nave91** [fishtown-analytics/dbt#3171]
### Describe the bug
Status is not accurate on nodes of `run_results.json`. The dbt model's error state is not represented accurately in `run_results.json`. 

### Steps To Reproduce
Caveating that I was not able to reproduce this locally with 0.19.0 version of dbt. I am not sure what configuration leads to this scenario. 
I've attached relevant artifacts below.
![image-1](https://user-images.githubusercontent.com/4484507/111380144-5bd8d080-867a-11eb-9b34-a8550f5a428b.png)

```
{
"status": "success",
"timing": [
{
"name": "compile",
"started_at": "2021-02-18 21:07:25.082433Z",
"completed_at": "2021-02-18 21:07:26.083771Z"
},
{
"name": "execute",
"started_at": "2021-02-18 21:07:26.085808Z",
"completed_at": "2021-02-18 21:07:26.085831Z"
}
],
"thread_id": "Thread-19",
"execution_time": 1.226057529449463,
"message": null,
"adapter_response": {},
"unique_id": "model.dbt_workspace.hard_deletes__augmented_snapshot_view"
}
```

### Expected behavior
I would expect model value in run_results.json to match output printed in stdout. Seems like they are differing. 

### Screenshots and log output
Attached below

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [x] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
v0.19.0

**The operating system you're using:**

**The output of `python --version`:**

### Additional context

```sql
-- target/run/hard_deletes__augmented_snapshot_view.sql
  create or replace  view ANALYTICS.dbt_cloud_pr_15782_12.hard_deletes__augmented_snapshot_view  as (
    with

    snapshot_with_distant_past_and_distant_future_raw as (
        select
            *,

            if(
                dbt_valid_from = min(dbt_valid_from) over (
            partition by id order by dbt_valid_from rows between unbounded preceding and unbounded following
        ),
                TIMESTAMP('1900-01-01 00:00:00+00'),
                dbt_valid_from
            ) as valid_from_with_distant_past,

            if(
                dbt_valid_to is null,
                TIMESTAMP('9999-12-31 23:59:59+00'),
                dbt_valid_to
            ) as valid_to_with_distant_future

        from ANALYTICS.dbt_cloud_pr_15782_12.hard_deletes__snapshot
    ),


    snapshot_with_distant_past_and_distant_future as (
        select 
            "ID",
  "COLOR",
  "THE_ANSWER",
  "INSERT_TIME",
  "DBT_SCD_ID",
  "DBT_UPDATED_AT",
  "DBT_VALID_FROM",
  "DBT_VALID_TO",
            valid_from_with_distant_past,
            valid_to_with_distant_future
        
        from snapshot_with_distant_past_and_distant_future_raw
    ),


    snapshot_with_is_deleted as (
        select 
            *,
            first_value(dbt_valid_to) over (
                partition by id order by dbt_valid_to asc rows between unbounded preceding and unbounded following
            ) is not null as is_deleted
        
        from snapshot_with_distant_past_and_distant_future
    ),

    snapshot_with_current_values as (
        select 
            *,

            if(
                is_deleted or (dbt_valid_to is null),
                true,
                false
            ) as is_current
        
        from snapshot_with_is_deleted
    )

    select * from snapshot_with_current_values


    


  );

```

**Comment by kwigley** [comment]
@nave91 just confirming, was this we `dbt run` or another command?

**Comment by nave91** [comment]
@kwigley This was with `dbt run`

**Comment by jtcohen6** [comment]
@nave91 I checked into the full `run_results.json` from the associated dbt Cloud job (thanks for sharing with me!), and I confirmed based on the `args` dict that this was the `run_results.json` artifact produced by the job's final `docs generate` step, rather than the earlier `run` step with the error:
```json
...
    "elapsed_time": 2.8335185050964355,
    "args": {
        "log_format": "default",
        "write_json": true,
        "profiles_dir": "/tmp/jobs/14862592/.dbt",
        "profile": "user",
        "target": "qa",
        "use_cache": true,
        "compile": true,
        "version_check": true,
        "which": "generate",
        "rpc_method": "docs.generate"
    }
...
```
This seems in line with the `execute` step taking 0.023 milliseconds, and the entire elapsed time being 2.8 seconds.

The issue here is that dbt overwrites artifacts, and the dbt Cloud run history page makes only the "last" version of each artifact available—though I understand that, behind the scenes, Cloud does a good job of saving the artifact from every run step. For jobs with "Generate docs?" checked, the final run step will always be `docs generate`, and it isn't skipped if a preceding step fails, so the `run_results.json` available for download will always reflect `docs generate` instead of the earlier failing step.

If we can locate the `run_results.json` artifact for the `dbt run` step of that job run, we should be able to confirm that `model.dbt_workspace.hard_deletes__augmented_snapshot_view` has `"status": "error"`.
